maven-plugin/src/main/java/org/openmrs/maven/plugins/Clone.java

```diff
@@ -0,0 +1,189 @@
+    }
+
+    private String extractGitHubHttpKeyFromModulePom(String artifactId, String version, String groupId) throws MojoExecutionException, IOException {
+        downloadModulePom(new Artifact(artifactId, version, groupId, "pom"));
+        String pomFileName = this.artifactId+"-"+version+".pom";
+        File pomDir = new File("pom/");
+        File pom = new File("pom/", pomFileName);
+        Model pomProperties = Project.loadProject(pomDir, pomFileName).getModel();
+        String url = pomProperties.getScm().getUrl();
+        pom.delete();
+        pomDir.delete();
+        return StringUtils.removeEnd(url, "/") + ".git";
+    }
+
+    private void downloadModulePom(Artifact artifact) throws MojoExecutionException {
+        MojoExecutor.Element[] artifactItems = new MojoExecutor.Element[1];
+        artifactItems[0] = artifact.toElement("pom/");
+        List<Element> configuration = new ArrayList<Element>();
+        configuration.add(element("artifactItems", artifactItems));
+        executeMojo(
+                plugin(
+                        groupId(SDKConstants.PLUGIN_DEPENDENCIES_GROUP_ID),
+                        artifactId(SDKConstants.PLUGIN_DEPENDENCIES_ARTIFACT_ID),
+                        version(SDKConstants.PLUGIN_DEPENDENCIES_VERSION)
+                ),
+                goal(GOAL_COPY),
+                configuration(configuration.toArray(new Element[0])),
+                executionEnvironment(mavenProject, mavenSession, pluginManager)
+        );
+    }
+
+    private void forkRepo(String repoToForkName) {
+
+        wizard.showMessage("Forking repository:");
+        String repoToForkOwner = "openmrs";
+        GitHubClient client = new GitHubClient();
+        System.out.print("- Client authentication");
+        client.setCredentials(githubUsername, githubPassword);
+        wizard.showMessage(" - DONE");
+        RepositoryService service = new RepositoryService();
+
+        System.out.print("- Service authentication");
+        service.getClient().setCredentials(githubUsername, githubPassword);
+        wizard.showMessage(" - DONE");
+        RepositoryId toBeForked = new RepositoryId(repoToForkOwner, repoToForkName);
+        try {
+            System.out.print("- Forking");
+            service.forkRepository(toBeForked);
+            wizard.showMessage(" - DONE");
+        } catch (IOException e) {
+            throw new IllegalStateException("Failed to fork repository \"" + repoToForkOwner + "/" + repoToForkName+ "\"", e);
+        }
+    }
+
+    private void cloneRepo(String upstreamHttpKey, String repoOwner) {
+
+        repoOwner = "/" + repoOwner + "/";
+
+        String originHttpKey = upstreamHttpKey.replace(repoOwner, "/" + githubUsername + "/");
+
+        String repoName = upstreamHttpKey.substring(
+                upstreamHttpKey.indexOf(repoOwner) + repoOwner.length(),
+                upstreamHttpKey.indexOf(".git")
+        );
+
+        if ("false".equals(testMode)) {
```

Comment:
Please add a method to AbstractTask isInTestMode(), I bet it will be needed elsewhere in the future.